### DISCUSSION TITLE
Streaming API billing

### DISCUSSION THREAD

**Discussion by sjpritchard** [speechly/speechly#310] [Q&A]
For the streaming API, are we charged for the entire duration of a gRPC stream, or only for the duration of speech that is processed between each SLUEvent.START and SLUEvent.STOP? I'm trying to decide if I can just keep the stream open indefinitely, or I should close it after each utterance.

**Comment by bigdatabaracus** [comment]
Hello @sjpritchard, thanks for reaching out! Yes, we do charge for the entire duration of a gRPC stream. Also there is a few minute timeout for the stream. In other words if you want an experience where the system is listening indefinitely, it's definitely advisable to deploy voice activity detection (VAD) in front of the gRPC stream. In fact, our browser client and out unity client come with options to use VAD. Using directly the gRPC api there is no VAD option, so you will need to implement it yourself. [Silero VAD](https://github.com/snakers4/silero-vad) is something we can recommend.

**Reply by sjpritchard** [reply]
Thanks. I'm using Push-to-Talk with my C++ client so no issues with only sending data when the user is speaking. So if I understand correctly, I should explicitly terminate the stream after each utterance. Also, what is the minimum billing increment (eg some providers charge per second, others per second with a minimum of 15 seconds, etc)

**Reply by bigdatabaracus** [reply]
@sjpritchard No, you do not need to explicitly terminate the stream. Our billing is based on start and stop. We meter by second. We will not charge for a say 3 second utterance a minimum of 15 seconds or anything stupid like that 😄

**Reply by sjpritchard** [reply]
Sorry for belabouring the point - I may not have asked my original question clearly and I'm still a little confused. If I open the gRPC stream, wait for 5 seconds, issue SLUEvent.START, talk for 5 seconds, issue SLUEvent.STOP, wait for 5 seconds, then close the gRPC stream, will I get charged for 5 seconds (the interval between START and STOP), or 15 seconds (the interval between the gRPC open and close.
